app/endpoints/raffle.py

```diff
@@ -0,0 +1,738 @@
+from app.utils.types.raffle_types import RaffleStatusType
+from app.utils.types.tags import Tags
+
+router = APIRouter()
+hyperion_raffle_logger = logging.getLogger("hyperion.raffle")
+
+
+@router.get(
+    "/tombola/raffles",
+    response_model=list[schemas_raffle.RaffleComplete],
+    status_code=200,
+    tags=[Tags.raffle],
+)
+async def get_raffle(
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member),
+):
+    """
+    Return all raffles
+    """
+    raffles = await cruds_raffle.get_raffles(db)
+    return raffles
+
+
+@router.post(
+    "/tombola/raffles",
+    response_model=schemas_raffle.RaffleSimple,
+    status_code=201,
+    tags=[Tags.raffle],
+)
+async def create_raffle(
+    raffle: schemas_raffle.RaffleBase,
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member_of(GroupType.soli)),
+):
+    """
+    Create a new raffle
+
+    **The user must be a member of the group soli to use this endpoint**
+    """
+    group = await cruds_groups.get_group_by_id(group_id=raffle.group_id, db=db)
+    if not group:
+        raise HTTPException(status_code=404, detail="Group not found")
+
+    db_raffle = models_raffle.Raffle(id=str(uuid.uuid4()), **raffle.dict())
+
+    try:
+        result = await cruds_raffle.create_raffle(raffle=db_raffle, db=db)
+        return result
+    except IntegrityError as error:
+        raise HTTPException(status_code=422, detail=str(error))
+
+
+@router.patch(
+    "/tombola/raffles/{raffle_id}",
+    status_code=204,
+    tags=[Tags.raffle],
+)
+async def edit_raffle(
+    raffle_id: str,
+    raffle_update: schemas_raffle.RaffleEdit,
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member_of(GroupType.soli)),
+):
+    """
+    Edit a raffle
+
+    **The user must be a member of the group soli to use this endpoint**
+    """
+
+    raffle = await cruds_raffle.get_raffle_by_id(raffle_id=raffle_id, db=db)
+    if not raffle:
+        raise HTTPException(status_code=404, detail="Raffle not found")
+
+    await cruds_raffle.edit_raffle(
+        raffle_id=raffle_id, raffle_update=raffle_update, db=db
+    )
+
+
+@router.delete(
+    "/tombola/raffles/{raffle_id}",
+    status_code=204,
+    tags=[Tags.raffle],
+)
+async def delete_raffle(
+    raffle_id: str,
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member_of(GroupType.soli)),
+):
+    """
+    Delete a raffle.
+
+    **The user must be a member of the group soli to use this endpoint**
+    """
+
+    raffle = await cruds_raffle.get_raffle_by_id(raffle_id=raffle_id, db=db)
+    if not raffle:
+        raise HTTPException(status_code=404, detail="Raffle not found")
+
+    await cruds_raffle.delete_raffle(raffle_id=raffle_id, db=db)
+
+
+@router.get(
+    "/tombola/group/{group_id}/raffles",
+    response_model=list[schemas_raffle.RaffleSimple],
```

Comment:
The endpoint will either return à `schemas_raffle.RaffleSimple` or `None` but not a `list[schemas_raffle.RaffleSimple]`